dspace-api/src/main/java/org/dspace/authenticate/ShibAuthentication.java

```diff
@@ -305,7 +298,10 @@ public int authenticate(Context context, String username, String password,
 			}
 
 			// Get the Shib supplied affiliation or use the default affiliation
-			List<String> affiliations = findMultipleAttributes(request, roleHeader);
+                        List<String> affiliations = new ArrayList<String>();
+                        if (context.getCurrentUser().getMetadata(roleHeader) != null) {
```

Comment:
After a bit of debugging I found out, that the request and the session are set, but the session is not containing the Shibboleth header or any shibboleth values. 

Here are the Request attributes I have in the getSpecialGroups() method:
javax.servlet.request.ssl_session
javax.servlet.request.ssl_session_id
dspace.current.user
javax.servlet.request.key_size
is.admin
javax.servlet.request.cipher_suite

And these are the attributes of the session at this point:
originatingHostIP: <my-ip>
resuming.request: true
dspace.current.remote.addr: <my-ip>
originatingHostName: <my-ip>
interrupted.request.info: org.dspace.app.webui.util.RequestInfo@43de46e4
dspace.current.user.id: 2
javax.servlet.jsp.jstl.fmt.locale.session: en
interrupted.request.url: https://<mydspaceurl>/mydspace

So the method obviously has no information about the Shibboleth header.